aten/src/ATen/native/mps/operations/LossOps.mm

```diff
@@ -873,132 +873,101 @@ void smooth_l1_loss_backward_impl(
     double beta,
     Tensor& grad_input)
 {
- struct CachedGraph : public MPSCachedGraph
-  {
+  if (grad_input.numel() == 0) {
+    return;
+  }
+  TORCH_CHECK(beta >= 0, "smooth_l1_loss_backward does not support negative values for beta.");
+
+  struct CachedGraph : public MPSCachedGraph {
     CachedGraph(MPSGraph *graph) : MPSCachedGraph(graph) {}
     MPSGraphTensor *inputTensor_ = nil;
     MPSGraphTensor *targetTensor_ = nil;
     MPSGraphTensor *gradInputTensor_ = nil;
+    MPSGraphTensor* gradOutputTensor_ = nil;
   };
 
- MPSGraphCache *cache_ = MPSGraphCache::getInstance();
-
-  MPSStream *stream= getCurrentMPSStream();
-
   @autoreleasepool {
+    string key = "smooth_l1_loss_backward" + getTensorsStringKey({input, grad_output, grad_input, target}) + ":"
```

Comment:
I don't think `grad_input` (the result is needed here) - this is always computed based on the input (which is already in the key)